docx/text/tabstops.py

```diff
@@ -21,6 +21,15 @@ class TabStops(ElementProxy):
 
     __slots__ = ()
 
+    def __getitem__(self, key):
+        """
+        Enables dictionary-style access by index.
+        """
+        tabs = self._element.tabs
+        if tabs is None:
+            return None
+        return TabStop(tabs[key])
+
```

Comment:
A couple problems here:

1. The docstring indicates "dictionary-style". I think we're talking about "list-style" behavior here. Along those same lines, I'd prefer `idx` rather than `key` to impress on the reader we expect a non-negative integer here.

2. `tabs` is only the parent element, it does not have sequence behaviors (despite the naming :). So the `w:tab` element in question here comes from `tabs.tab_lst[key]`, not `tabs[key]`. I'd put that access on a separate line just to keep things easy reading. Also, it produces a marginally more useful error message when someone chooses an index out of range.

3. There are no cases here where indexed access should return `None`. It should return a tab stop or raise an exception. In an empty TabStops object for example, even tab_stops[0] is invalid. The exception should be `IndexError`.